scarb/src/core/registry/client/http.rs

```diff
@@ -146,12 +151,77 @@ impl<'c> RegistryClient for HttpRegistryClient<'c> {
     }
 
     async fn supports_publish(&self) -> Result<bool> {
-        // TODO(mkaput): Publishing to HTTP registries is not implemented yet.
-        Ok(false)
+        let index_config = self.index_config.load().await?;
+        match index_config.upload {
+            Some(_) => Ok(true),
+            None => Ok(false),
+        }
     }
 
-    async fn publish(&self, _package: Package, _tarball: FileLockGuard) -> Result<()> {
-        todo!("Publishing to HTTP registries is not implemented yet.")
+    async fn publish(&self, package: Package, tarball: FileLockGuard) -> Result<RegistryUpload> {
+        let auth_token = env::var("SCARB_REGISTRY_AUTH_TOKEN")
```

Comment:
is this temporary?

Reply:
1. clap support env vars
2. I asked whether this is temporary because expecting people to set env manually is bad UX imho

Reply:
Hmm, ok I thought about it from devops and security pov - I wouldn't ever want my tokens or passwords to be passed as flags to any program. Just having this as plain text in bash_history is bad enough, so I actually disagree with you on that points. They are valid in most cases, but not sure in this particular one.

Reply:
Maybe this is an idea for a "feature"? :) 
regarding exports etc, yup this still leaks stuff. But what I meant is more of an "automatic" env, like some VMs created with env variables, CIs etc :+1:

Reply:
> Maybe this is an idea for a "feature"? :)

Sure, just an idea to address both your and Marek's concerns, i.e. to both have no leaks and good UX. Added #1579 to further discuss this.

> But what I meant is more of an "automatic" env, like some VMs created with env variables, CIs etc 👍

Yeah, I understand, make sense

Reply:
I meant, you can expect env vars in clap arguments without exposing them as parameters, just don't use `short` nor `long` in relevant `#[arg]` use

Reply:
Sure, I think this is a bit misleading then

> I asked whether this is temporary because expecting people to set env manually is bad UX imho

as users will still have to set this manually, right? so actually nothing changes from UX perspective?